Lib/Config/RedisConfig.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Emoncms\Config;
+
+class RedisConfig
+{
+    /**
+     * @var string
+     */
+    private $host;
+
+    /**
+     * @var int
+     */
+    private $port;
+
+    /**
+     * @var string
+     */
+    private $auth;
+
+    /**
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @var bool
+     */
+    private $isRedisEnabled;
+
+    /**
+     * @param bool $isRedisEnabled
+     * @param string $host
+     * @param int $port
+     * @param string $auth
+     * @param string $prefix
+     */
+    public function __construct($isRedisEnabled, $host, $port, $auth, $prefix)
+    {
+        $this->isRedisEnabled = $isRedisEnabled;
```

Comment:
Ideally we should be able to drop this if the other settings are empty, but baby steps :dancer: